canvasapi/canvas.py

```diff
@@ -437,23 +437,32 @@ def get_activity_stream_summary(self, **kwargs):
         )
         return response.json()
 
-    def get_announcements(self, **kwargs):
+    def get_announcements(self, course_list, **kwargs):
         """
         List announcements.
 
         :calls: `GET /api/v1/announcements \
         <https://canvas.instructure.com/doc/api/announcements.html#method.announcements_api.index>`_
 
+        :param course_list: Course ID(s) to request announcements from.
+        :type course_list: list
+
         :rtype: :class:`canvasapi.paginated_list.PaginatedList` of
                 :class:`canvasapi.discussion_topic.DiscussionTopic`
         """
         from canvasapi.discussion_topic import DiscussionTopic
 
+        if type(course_list) is not list:
+            raise RequiredFieldMissing("Course IDs need to be passed as a list")
+
+        course_ids = [f"course_{course_id}" for course_id in course_list]
+
         return PaginatedList(
             DiscussionTopic,
             self.__requester,
             "GET",
             "announcements",
+            context_codes=course_ids,
```

Comment:
Discussion for how to handle `context_codes` is ongoing, but regardless of what decision we come to, `context_codes` will almost certainly be a `list`. For our parameter "unpacking"/"combining" to work, it will need to be run through `combine_kwargs`.

The best way to do that is to add `context_codes` into `kwargs` before the `combine_kwargs` call.

Something like this, before the PaginatedList object:
```python
kwargs['context_codes'] = course_ids
```